src/status_im/contexts/wallet/events.cljs

```diff
@@ -327,10 +327,16 @@
  :wallet/set-ens-address
  (fn [{:keys [db]} [result ens]]
    (let [suggestion (if result
-                      [{:type     item-types/address
-                        :ens      ens
-                        :address  (eip55/address->checksum result)
-                        :networks [:ethereum :optimism]}]
+                      [{:type         item-types/address
+                        :ens          ens
+                        :address      (eip55/address->checksum result)
+                        :networks     [constants/ethereum-network-name constants/optimism-network-name]
+                        :full-address (str (network-utils/short-names->network-preference-prefix
+                                            [(get network-utils/network->short-name
```

Comment:
Nice improvement @mmilad75. What follows is just a light suggestion, something I'd do. Maybe you like the overall idea, even if you don't want to apply it now.

This part can be more general/functional and you can reuse the sequence of networks too. While thinking about this suggestion, it forced me to think about what `result` really means as a word. The goal in this refactor was to use the principle of "tell don't ask", but applied to a functional paradigm.

> Tell-Don't-Ask is a principle that helps people remember that object-orientation is about bundling data with the functions that operate on that data. It reminds us that rather than asking an object for data and acting on that data, we should instead tell an object what to do.
>
> -- https://martinfowler.com/bliki/TellDontAsk.html 

Translated to our case here, it's like *rather than asking for individual pieces of data and assembling them together, make that a verb (function) and tell it to do its thing*.

```clojure
(defn- resolved-address->prefixed-address
  [address networks]
  (let [prefixes (->> networks
                      (map network-utils/network->short-name)
                      network-utils/short-names->network-preference-prefix)]
    (str prefixes (eip55/address->checksum address))))

(rf/reg-event-fx :wallet/set-ens-address
 (fn [{:keys [db]} [result ens]]
   (let [networks   [constants/ethereum-network-name constants/optimism-network-name]
         suggestion (if result
                      [{:type         item-types/address
                        :ens          ens
                        :address      (eip55/address->checksum result)
                        :networks     networks
                        :full-address (resolved-address->prefixed-address result networks)}]
                      [])]
     {:db (-> db
              (assoc-in [:wallet :ui :search-address :local-suggestions] suggestion)
              (assoc-in [:wallet :ui :search-address :valid-ens-or-address?] (boolean result)))})))

```

Reply:
This is a very great suggestion, thanks man. Every time I learn a lot from you 🙏 @ilmotta

Reply:
Done